website/docs/r/logs_pipeline.html.markdown

```diff
@@ -0,0 +1,254 @@
+            preserve_source = true
+            override_on_conflict = false
+        }
+    }
+    processor {
+        category_processor {
+            name = "sample category processor"
+            is_enabled = true
+            target = "foo.severity"
+            category {
+                name = "debug"
+                filter {
+                    query = "@severity: \".\""
+                }
+            }
+            category {
+                name = "verbose"
+                filter {
+                    query = "@severity: \"-\""
+                }
+            }
+        }
+    }
+    processor {
+        date_remapper {
+            name = "sample date remapper"
+            is_enabled = true
+            sources = ["_timestamp", "published_date"]
+        }
+    }
+    processor {
+        message_remapper {
+            name = "sample message remapper"
+            is_enabled = true
+            sources = ["msg"]
+        }
+    }
+    processor {
+        status_remapper {
+            name = "sample status remapper"
+            is_enabled = true
+            sources = ["info", "trace"]
+        }
+    }
+    processor {
+        trace_id_remapper {
+            name = "sample trace id remapper"
+            is_enabled = true
+            sources = ["dd.trace_id"]
+        }
+    }
+    processor {
+        service_remapper {
+            name = "sample service remapper"
+            is_enabled = true
+            sources = ["service"]
+        }
+    }
+    processor {
+        grok_parser {
+            name = "sample grok parser"
+            is_enabled = true
+            source = "message"
+            grok {
+                support_rules = ""
+                match_rules = "Rule %%{word:my_word2} %%{number:my_float2}"
+            }
+        }
+    }
+    processor {
+        pipeline {
+            name = "nested pipeline"
+            is_enabled = true
+            filter {
+                query = "source:foo"
+            }
+            processor {
+                url_parser {
+                    name = "sample url parser"
+                    sources = ["url", "extra"]
+                    target = "http_url"
+                    normalize_ending_slashes = true
+                }
+            }
+        }
+    }
+    processor {
+        user_agent_parser {
+            name = "sample user agent parser"
+            is_enabled = true
+            sources = ["user", "agent"]
+            target = "http_agent"
+            is_encoded = false
+        }
+    }
+}
+```
+## Example Usage: Datadog logs pipeline order
+
+```hcl
+resource "datadog_logs_pipelineorder" "sample_pipeline_order" {
+    name = "sample_pipeline_order"
+    depends_on = [
+        "datadog_logs_pipeline.sample_pipeline"
+    ]
+    pipelines = [
+        "${datadog_logs_pipeline.my.id}"
+    ]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Your pipeline name.
```

Comment:
Why don't we group required parameters and put all the optional ones at the end ?

Reply:
I will do that.

Reply:
I agree with @tt810 here, but I honestly don't have a strong opinion.

Reply:
I'm more inclined to have required first and optional following as when you create the resource it's more natural to have the required elements first otherwise you'd have to scan all the attributes to ensure you did not miss any. You correctly have a below section with an anchor link that should help readers.

Reply:
Cf dashboard for example: https://www.terraform.io/docs/providers/datadog/r/dashboard.html

Reply:
Could we group required parameters and follow you convention, e.g. small fields first so that it makes everyone happy please ?

Reply:
I update the order to have required before optional.